pkgs/tools/text/dirdiff/default.nix

```diff
@@ -0,0 +1,55 @@
+{ fetchurl, lib, stdenv, tcl, tk
+, copyDesktopItems, makeDesktopItem
+}:
+stdenv.mkDerivation {
+  pname = "dirdiff";
+  version = "2.1";
+
+  src = fetchurl {
+    url = "https://www.samba.org/ftp/paulus/dirdiff-2.1.tar.gz";
+    sha256 = "0lljd8av68j70733yshzzhxjr1lm0vgmbqsm8f02g03qsma3cdyb";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems ];
+  buildInputs = [ tcl tk ];
+
+  # Some light path patching.
+  inherit tcl tk;
+  patches = [ ./dirdiff-2.1-vars.patch ];
+  postPatch = ''
+    for file in dirdiff Makefile; do
+      substituteInPlace "$file" \
+          --subst-var out --subst-var tcl --subst-var tk
+    done
+  '';
+
+  # If we don't create the directories ourselves, then 'make install' creates
+  # files named 'bin' and 'lib'.
+  preInstall = ''
+    mkdir -p $out/bin $out/lib
+  '';
+  installFlags = [ "BINDIR=\${out}/bin" "LIBDIR=\${out}/lib" ];
```

Comment:
```suggestion
  installFlags = [ "BINDIR=${placeholder "out"}/bin" "LIBDIR=${placeholder "out"}/lib" ];
```

Reply:
Interesting, I hadn't seen this magic function, thanks!  In grepping Nixpkgs it seems like `$(out)` is the preferred syntax for interpolating into `installFlags`, do you mind if I use that?

Reply:
I would prefer if you use placeholder. Greping nixpkgs is not a good metric because it also counts all the dust and rusted packages.

Reply:
Okay, I see that this is new norm, done.  I wish the Nix(pkgs?) manual documented when these should be used; I admit I'm not totally sure why variable substitution isn't a good idea here :).